sync3/caches/user.go

```diff
@@ -281,7 +282,7 @@ func (c *UserCache) OnRegistered(ctx context.Context, _ int64) error {
 		if !ok {
 			urd = NewUserRoomData()
 		}
-		urd.JoinNID = joinNIDs[room.RoomID]
+		urd.JoinTiming = joinTimings[room.RoomID]
```

Comment:
Is it possible for there to be no join timings and hence `urd.JoinTiming == nil`? Should init an empty map then?

Reply:
Good question. For this to be the case, the `joinedRooms` and `joinTimings` returned by `GlobalCache.LoadJoinedRooms` would have to return maps with different sets of keys. I would hope this is impossible---will check and leave a comment if so.

Reply:
Oh, I misunderstood you. You're talking about the entire `joinTimings` map being nil.

Reply:
> Is it possible for there to be no join timings and hence urd.JoinTiming == nil?

No, I've traced through the logic and concluded that LoadJoinedRooms either returns `(nil, nil, err)` or `(x, y, nil)` where x and y are non-nil maps with the same set of keys. I've left comments asserting this.